### PR TITLE
BufferHelpers added to the toolkit

### PR DESCRIPTION
**Pull request by walbourn**
Added a new header with ``CreateStaticBuffer`` and ``CreateTextureFromMemory``

### CODE REVIEW

**Comment by walbourn** [comment]
The test suite updates are in the [pull request](https://github.com/walbourn/directxtk12test/pull/19)

File: Src/BufferHelpers.cpp [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    return S_OK;
+}
+
+
+//--------------------------------------------------------------------------------------
+_Use_decl_annotations_
+HRESULT DirectX::CreateTextureFromMemory(
+    ID3D12Device* device,
+    ResourceUploadBatch& resourceUpload,
+    size_t width,
```
Comment (darkscott) [inline Src/BufferHelpers.cpp:87]: Should this just be a uint32_t?  I noticed in other places you changed from size_t to uint32_t... so maybe it makes sense to do that here, too?

File: Src/BufferHelpers.cpp [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    if (generateMips)
+    {
+        generateMips = resourceUpload.IsSupportedForGenerateMips(format);
+        if (generateMips)
+        {
+            mipCount = static_cast<uint16_t>(LoaderHelpers::CountMips(static_cast<uint32_t>(width), static_cast<uint32_t>(height)));
+        }
+    }
+
+    auto desc = CD3DX12_RESOURCE_DESC::Tex2D(format, static_cast<UINT64>(width), static_cast<UINT>(height), 1u, mipCount);
```
Comment (darkscott) [inline Src/BufferHelpers.cpp:180]: If Tex2D accepts a UINT64, why is there a check above ensuring it's < UINT32_MAX?

File: Src/BufferHelpers.cpp [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+    if ((width > D3D12_REQ_TEXTURE3D_U_V_OR_W_DIMENSION) || (width > UINT32_MAX)
+        || (height > D3D12_REQ_TEXTURE3D_U_V_OR_W_DIMENSION) || (height > UINT32_MAX)
+        || (depth > D3D12_REQ_TEXTURE3D_U_V_OR_W_DIMENSION) || (depth > UINT32_MAX))
+    {
+        DebugTrace("ERROR: Resource dimensions too large for DirectX 12 (3D: size %zu by %zu by %zu)\n", width, height, depth);
+        return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
+    }
+
+    auto desc = CD3DX12_RESOURCE_DESC::Tex3D(format, static_cast<UINT64>(width), static_cast<UINT>(height), static_cast<UINT16>(depth), 1u);
```
Comment (darkscott) [inline Src/BufferHelpers.cpp:248]: Same question here.. these casts don't make sense given the checks above... depth, especially, should check against UINT16_MAX, if anything..

**Pull request merged**